include/zephyr/usb/class/midi.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2024 Titouan Christophe
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_USB_CLASS_MIDI_H_
+#define ZEPHYR_INCLUDE_USB_CLASS_MIDI_H_
+
+/**
+ * @brief Universal MIDI Packet definitions
+ * @defgroup usb_midi_ump USB MIDI 2.0 Universal MIDI Packet definitions
+ * @ingroup usb
+ * @since 4.1
+ * @version 0.1.0
+ * @see ump112: "Universal MIDI Packet (UMP) Format and MIDI 2.0 Protocol"
+ *              Document version 1.1.2
+ * @{
+ *
+ * @defgroup usb_midi_ump_mt Message types
+ * @ingroup usb_midi_ump
+ * @see ump112: 2.1.4 Message Type (MT) Allocation
+ * @{
+ */
+
+#define UMP_MT_UTILITY             0x00 /**< Utility Messages */
+/** System Real Time and System Common Messages (except System Exclusive) */
+#define UMP_MT_SYS_RT_COMMON       0x01
+#define UMP_MT_MIDI1_CHANNEL_VOICE 0x02 /**< MIDI 1.0 Channel Voice Messages */
+#define UMP_MT_DATA                0x03 /**< Data Messages (including System Exclusive) */
+#define UMP_MT_MIDI2_CHANNEL_VOICE 0x04 /**< MIDI 2.0 Channel Voice Messages */
+#define UMP_MT_FLEX_DATA           0x0d /**< Flex Data Messages */
+#define UMP_MT_UMP_STREAM          0x0f /**< UMP Stream Message */
+/** @} */
+
+/**
+ * @brief      Message Type field of a Universal MIDI Packet
+ * @param[in]  ump    The universal MIDI Packet
+ */
+#define UMP_MT(ump)    ((ump)[0] >> 28)
+
+/**
+ * @brief      Size of a Universal MIDI Packet, in 32bit words
+ * @param[in]  ump    The universal MIDI Packet
+ * @see ump112: 2.1.4 Message Type (MT) Allocation
+ */
+#define UMP_WORDS(ump) (1 + ((0xfe950d40 >> (2 * UMP_MT(ump))) & 3))
```

Comment:
is there a name for the magic number here? (interesting expression btw, I think it may use a comment)

Reply:
There are 16 message types in UMP, and their length vary between 1 and 4 words; so this is actually an array of 16x2bits numbers packed together as an uint32 :sweat_smile:
In other words, this is not a magic number/constant from the spec, but rather a memory-packed lookup table.

I agree this is worth a comment, will do !

Reply:
oh, fancy, would it make sense to build the number up in a couple macros with the value and bit shift then? That would probably make it obvious (or at least understandable) without needing a comment in the first place